resharper/src/resharper-unity/CSharp/Feature/Services/Daemon/UnityObjectIncorrectBaseTypeErrorHandler.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+using JetBrains.ReSharper.Feature.Services.Daemon;
+using JetBrains.ReSharper.Plugins.Unity.CSharp.Daemon.Errors;
+using JetBrains.ReSharper.Plugins.Unity.CSharp.Psi.Resolve;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.CSharp;
+using JetBrains.ReSharper.Psi.Resolve;
+
+namespace JetBrains.ReSharper.Plugins.Unity.CSharp.Feature.Services.Daemon
+{
+    [Language(typeof(CSharpLanguage))]
+    public class UnityObjectIncorrectBaseTypeErrorHandler : IResolveProblemHighlighter
+    {
+        // We only get called for our known resolve errors
+        public IHighlighting Run(IReference reference)
+        {
+            var errorType = reference.CheckResolveResult();
+            if (errorType == UnityResolveErrorType.EXPECTED_COMPONENT)
```

Comment:
I don't see any warning on my Mac OS machine, only highlighting (no tooltip). I will check it tomorrow on Windows machine.
In inspection result the correct message is presented.